jenkins-s2i-build/jenkins-s2i-build-template-with-git-secret-and-image-pull-secret.yml

```diff
@@ -0,0 +1,157 @@
+---
+kind: Template
+apiVersion: template.openshift.io/v1
+metadata:
+  name: jenkins-s2i
+  annotations:
+    openshift.io/display-name: Jenkins S2I
+    description: Jenkins S2I build config to create a Jenkins image with your configuration
+      baked in.
+    iconClass: icon-jenkins
+    tags: instant-app,jenkins
+objects:
+- apiVersion: v1
+  stringData:
+    password: "${SOURCE_REPOSITORY_PASSWORD}"
+    username: "${SOURCE_REPOSITORY_USERNAME}"
+  kind: Secret
+  metadata:
+    name: git-auth
+  type: kubernetes.io/basic-auth
+- kind: ImageStream
+  apiVersion: image.openshift.io/v1
+  metadata:
+    name: "${NAME}"
+    annotations:
+      description: Keeps track of changes in the application image
+- kind: ImageStream
+  apiVersion: image.openshift.io/v1
+  metadata:
+    name: ${BUILDER_IMAGE_STREAM_NAME}
+    annotations:
+      description: Tracks the Red Hat Jenkins base image, since openshift may not ship with the tag we need. Importing the image allows normal users to run the playbook.
+  spec:
+    tags:
+    - annotations: null
+      from:
+        kind: DockerImage
+        name: ${BUILDER_REGISTRY_URL}/${BUILDER_REPOSITORY_NAME}/${BUILDER_IMAGE}:${BUILDER_IMAGE_STREAM_TAG_NAME}
+      importPolicy: {}
+      name: ${BUILDER_IMAGE_STREAM_TAG_NAME}
+      referencePolicy:
+        type: Source
+- kind: BuildConfig
+  apiVersion: build.openshift.io/v1
+  metadata:
+    name: "${NAME}"
+    labels:
+      build: "${NAME}"
+      type: image
+  spec:
+    triggers:
+    - type: GitHub
+      github:
+        secret: "${BUILD_SECRET}"
+    - type: GitLab
+      gitlab:
+        secret: "${BUILD_SECRET}"
+    - type: BitBucket
+      bitbucket:
+        secret: "${BUILD_SECRET}"
+    - type: Generic
+      generic:
+        secret: "${BUILD_SECRET}"
+        allowEnv: true
+    - type: ConfigChange
+    runPolicy: Serial
+    source:
+      type: Git
+      sourceSecret:
+        name: "git-auth"
+      git:
+        uri: "${SOURCE_REPOSITORY_URL}"
+        ref: "${SOURCE_REPOSITORY_REF}"
+      contextDir: "${SOURCE_REPOSITORY_CONTEXT_DIR}"
+    strategy:
+      type: Source
+      sourceStrategy:
+        env:
+          - name: "GIT_SSL_NO_VERIFY"
+            value: "true"
```

Comment:
Can we make this a template param, and just default it to `true`?

Reply:
done